src/components/createupdatevariable.vue

```diff
@@ -0,0 +1,823 @@
+      );
+    });
+
+    const validateSecretName = (secret: Secret) => {
+      secret.error = '';
+      if (!secret.name) {
+        secret.error = 'Secret name is required.';
+      }
+    };
+
+    const validateSecretValue = (secret: Secret) => {
+      secret.error = '';
+      if (!secret.value) {
+        secret.error = 'Secret value is required.';
+      }
+    };
+
+    const areConditionValuesValid = computed(() => {
+      return secrets.value.every((secret) => {
+        return secret.conditions.every((condition) => {
+          if (
+            condition.type === 'branch' ||
+            condition.type === 'tag' ||
+            condition.type === 'ref'
+          ) {
+            return (
+              condition.conditionType &&
+              condition.actions &&
+              condition.actions.length > 0 &&
+              condition.actions.every((action) => {
+                return action.action !== '' && action.input !== '';
+              })
+            );
+          } else {
+            return true;
+          }
+        });
+      });
+    });
+
+    // disable condition fields when condition type is not defined
+    const isConditionFieldDisabled = (condition: ConditionSecret) => {
+      return !condition.type;
+    };
+
+    const resetConditionProperties = (conditions: ConditionSecret[]) => {
+      conditions.forEach((condition) => {
+        if (!condition.type) {
+          condition.conditionType = '';
+          condition.actions = [{ action: '', input: '' }];
+        }
+      });
+    };
+
+    const isRegexMatchValid = computed(() => {
+      return secrets.value.every((secret) => {
+        return secret.conditions.every((condition) => {
+          return condition.actions.every((action) => {
+            return condition.type === 'regex' && isValidRegex(action.input);
+          });
+        });
+      });
+    });
+
+    function isValidRegex(input: string) {
+      try {
+        new RegExp(input);
+        return true;
+      } catch (e) {
+        return false;
+      }
+    }
+
+    const handleActionChange = (condition: ConditionSecret) => {
+      condition.actions.forEach((action) => {
+        if (action.action === 'regex' && !isRegexMatchValid.value) {
+          regexMatchErrorMessage.value = 'Invalid regular expression pattern.';
+        }
+        if (action.action === 'simple' && regexMatchErrorMessage.value) {
+          regexMatchErrorMessage.value = '';
+        }
+      });
+    };
+
+    const atLeastOneSecretErrorMessage = computed(() => {
+      if (isSubmitted.value && !isAtLeastOneSecretValid.value) {
+        return 'At least one secret name and value is required.';
+      }
+      return '';
+    });
+
+    const conditionValuesErrorMessage = computed(() => {
+      if (!areConditionValuesValid.value) {
+        return 'All condition fields must be filled for each secret condition.';
+      }
+      return '';
+    });
+
+    const isFormValid = computed(() => {
```

Comment:
use the same logic of createupdatesecrets.vue (the name is isSaveButtonDisabled)